### PR TITLE
174264646  Set logging levels via ENV

### PR DESCRIPTION
**Pull request by knowuh**
There is so much logging going to STDOUT by default.

It makes it hard to read test status.

Developers can configure logging verbosity via .env file.

[#174264646 ]

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -104,9 +104,14 @@ services:
       DOCKER: 'true'
 
       #
-      # Enables logging to stdout instead of file.
+      # Enables logging to stdout instead of file. Override this in .env
       #
-      RAILS_STDOUT_LOGGING: "true"
+      RAILS_STDOUT_LOGGING: "${RAILS_STDOUT_LOGGING:-true}"
```
Comment (scytacki) [inline docker-compose.yml:109]: This line change shouldn't be necessary. Any ENV defined in docker-compose can be overridden by the .env.
These kinds of defaults inside of `${...}` are only necessary when the variable is used in other parts of the file (not the environment section.

File: docker-compose.yml [hunk 2]
```diff
@@ -104,9 +104,14 @@ services:
...
+
+      # Configure rails logging level. Override this in .env
+      # Rails logging levels:
+      # DEBUG | INFO | WARN | ERROR | FATAL | UNKNOWN
+      LOG_LEVEL: "${LOG_LEVEL:-DEBUG}"
```
Comment (scytacki) [inline docker-compose.yml:114]: Same with this. I should just be:
`LOG_LEVEL: "DEBUG"`

File: docker-compose.yml (hunk 1 above)
Comment (scytacki) [reply docker-compose.yml:109]: Never mind I was wrong. This is needed.

File: docker-compose.yml (hunk 2 above)
Comment (scytacki) [reply docker-compose.yml:114]: Never mind I was wrong. This is needed.

File: rails/config/environments/cucumber.rb [hunk 3]
```diff
@@ -41,4 +41,9 @@
     # Disable logging to file. It might have performance impact while using Docker for Mac (slow filesystem sync).
     config.logger = Logger.new(STDOUT)
   end
+  # Specify developer chosen logging level. See docker-compose.yml and .env
+  # Using to something other than (0) debug makes test output more readable.
+  if (log_level = ENV['TEST_LOG_LEVEL'])
```
Comment (scytacki) [inline rails/config/environments/cucumber.rb:46]: Just realized this kind of comparison will probably fail if someone sets the ENV variable to an empty string. 
So it'd probably be better to change it to:
```
log_level = ENV['TEST_LOG_LEVEL']
if(log_level.present?)
  ...
```

**Comment by knowuh** [comment]
@scytacki  I think f6e7131 addresses your last concern.

**Comment by knowuh** [comment]
… and now test are passing too.  👍

File: rails/config/environments/cucumber.rb (hunk 3 above)
Comment (knowuh) [reply rails/config/environments/cucumber.rb]: f6e7131 addresses this.

**Comment by scytacki** [comment]
I think it is fine to merge as is, but I think it will have some complex behavior:
- if the ENV variable is not defined at all then the default from fetch is used
- if the ENV variable is defined to be the empty string `""` then the log level will not be modified

It'd be better for those two cases to result in the same thing. The issue is that as far as I can tell fetch doesn't handle the empty string case.  If you use ENV['var'], and then use `present?` or `blank?` depending on how you want to handle the default, will make the two cases above behave the same way.

**Comment by knowuh** [comment]
c11a978 handles unset, blank, or incorrectly set ENV values for TEST_LOG_LEVEL and DEV_LOG_LEVEL consistently, and adds a small spec test to verify this behavior.

**Pull request merged**